### PR TITLE
feat(frontend): avoid pk duplication 

### PR DESCRIPTION
**Pull request by wcy-fdu**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

avoid pk duplication for streaming executors, and still allow join key duplication.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

None

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

## Refer to a related PR or issue link (optional)
close https://github.com/risingwavelabs/risingwave/issues/6784

### CODE REVIEW

**Review by st1page — APPROVED** [review]

File: src/frontend/src/optimizer/plan_node/generic/agg.rs [hunk 1]
```diff
@@ -211,8 +215,11 @@ impl<PlanRef: stream::StreamPlanRef> Agg<PlanRef> {
             let mut included_upstream_indices = vec![];
             for &idx in &self.group_key {
                 let tb_column_idx = internal_table_catalog_builder.add_column(&in_fields[idx]);
-                internal_table_catalog_builder
-                    .add_order_column(tb_column_idx, OrderType::Ascending);
+                internal_table_catalog_builder.add_order_column(
+                    tb_column_idx,
+                    OrderType::Ascending,
+                    true,
+                );
```
Comment (yuhao-su) [inline src/frontend/src/optimizer/plan_node/generic/agg.rs:222]: We  can't dedup group key like this

File: src/frontend/src/optimizer/plan_node/generic/top_n.rs [hunk 2]
```diff
@@ -58,21 +58,24 @@ impl<PlanRef: stream::StreamPlanRef> TopN<PlanRef> {
         // does a prefix scanning with the group key, we can fetch the data in the
         // desired order.
         self.group_key.iter().for_each(|&idx| {
-            internal_table_catalog_builder.add_order_column(idx, OrderType::Ascending);
+            internal_table_catalog_builder.add_order_column(idx, OrderType::Ascending, true);
```
Comment (yuhao-su) [inline src/frontend/src/optimizer/plan_node/generic/top_n.rs:61]: ditto

File: src/frontend/src/optimizer/plan_node/stream_materialize.rs [hunk 3]
```diff
@@ -91,7 +91,7 @@ impl StreamMaterialize {
         let input = required_dist.enforce_if_not_satisfies(input, &Order::any())?;
         let base = PlanBase::derive_stream_plan_base(&input);
         let schema = &base.schema;
-        let pk_indices = &base.logical_pk;
+        let pk_indices = &base.logical_pk.iter().copied().unique().collect_vec();
```
Comment (yuhao-su) [inline src/frontend/src/optimizer/plan_node/stream_materialize.rs:94]: Does this include order by columns? If yes, we can't dedupe

File: src/frontend/src/optimizer/plan_node/generic/agg.rs (hunk 1 above)
Comment (st1page) [reply src/frontend/src/optimizer/plan_node/generic/agg.rs:222]: why？

File: src/frontend/src/optimizer/plan_node/stream_materialize.rs (hunk 3 above)
Comment (st1page) [reply src/frontend/src/optimizer/plan_node/stream_materialize.rs:94]: I think not, the `user_order_by` is order-dependent

**Comment by yuhao-su** [comment]
Currently there might be an inconsistency between the pk of hash join state table and the input pk got in hash join executor. We need to do more refactor than this to complete the dedup task.  I suggest we reverting this PR for now.

**Pull request merged**